source/slang/slang-check-overload.cpp

```diff
@@ -1013,6 +1013,29 @@ namespace Slang
             }
         }
 
+        // If both parents are the same, there is no explicit overload preference without looking
+        // at parameters, which is resolved later.
+        if(left.declRef.getParent() == right.declRef.getParent())
+            return 0;
+
+        // If both are members of an inherited decl, prefer the more derived inheritance member
+        auto leftStruct = left.declRef.getParent().as<StructDecl>();
+        auto rightStruct = right.declRef.getParent().as<StructDecl>();
+        if (leftStruct && rightStruct)
+        {
+            auto leftType = DeclRefType::create(m_astBuilder, leftStruct);
```

Comment:
This step drops the generic substitution arguments from the parent type. DeclRef::getParent should give you a decl ref to the parent that has the generic arguments in the case if a generic specialization, we need to create a DeclRef type from that full DeclRef instead of just a decl.

Reply:
I will add a test for this as well (override across generics)